src/compiler/chainrules.jl

```diff
@@ -123,11 +123,33 @@ Convert `x` from the format Zygote uses internally to differentials types ChainR
 """
 @inline wrap_chainrules_input(x) = x
 @inline wrap_chainrules_input(::Nothing) = ChainRules.ZeroTangent()
+@inline wrap_chainrules_input(::AbstractArray{Nothing}) = ChainRules.ZeroTangent()
```

Comment:
This takes us another step closer to #1057 
(but avoids the worst cases of it) still

Reply:
Ok. But this should have no bad side effects right? No extra memory, no conflict with doing #1057 later?

Reply:
Correct.